// vitepress-theme-teek/lib/helper/is.js
'use strict';

const isExternal = (path) => /^(http?:|https?:|mailto:|tel:)/.test(path);
const isValidURL = (url) => {
  try {
    new URL(url);
    return true;
  } catch {
    return false;
  }
};
const isType = (val) => {
  if (val === null) return "null";
  if (typeof val !== "object") return typeof val;
  else return Object.prototype.toString.call(val).slice(8, -1).toLocaleLowerCase();
};
const is = (val, type) => {
  return Object.prototype.toString.call(val) === `[object ${type}]`;
};
const isFunction = (val) => {
  return is(val, "Function");
};
const isDef = (val) => {
  return typeof val !== "undefined";
};
const isUnDef = (val) => {
  return !isDef(val);
};
const isObject = (val) => {
  return val !== null && is(val, "Object");
};
const isDate = (val) => {
  return is(val, "Date");
};
const isNumber = (val) => {
  const regPos = /^\d+(\.\d+)?$/;
  const regNeg = /^(-(([0-9]+\.[0-9]*[1-9][0-9]*)|([0-9]*[1-9][0-9]*\.[0-9]+)|([0-9]*[1-9][0-9]*)))$/;
  if (regPos.test(val) || regNeg.test(val)) {
    return true;
  } else {
    return false;
  }
};
const isStringNumber = (val) => {
  if (!isString(val)) return false;
  return !Number.isNaN(Number(val));
};
const isAsyncFunction = (val) => {
  return is(val, "AsyncFunction");
};
const isPromise = (val) => {
  return is(val, "Promise") && isObject(val) && isFunction(val.then) && isFunction(val.catch);
};
const isString = (val) => {
  return is(val, "String");
};
const isBoolean = (val) => {
  return is(val, "Boolean");
};
const isArray = (arg) => {
  if (typeof Array.isArray === "undefined") {
    return Object.prototype.toString.call(arg) === "[object Array]";
  }
  return Array.isArray(arg);
};
const isClient = typeof window !== "undefined" && typeof document !== "undefined";
const isServer = !isClient;
const inBrowser = isClient;
const isElement = (val) => {
  if (typeof Element === "undefined") return false;
  return val instanceof Element;
};
const isImageDom = (o) => {
  return o && ["IMAGE", "IMG"].includes(o.tagName);
};
const isNull = (val) => {
  return val === null;
};
const isNullAndUnDef = (val) => {
  return isUnDef(val) && isNull(val);
};
const isNullOrUnDef = (val) => {
  return isUnDef(val) || isNull(val);
};
const isPhone = (val) => {
  return /^(13[0-9]|14[01456879]|15[0-35-9]|16[2567]|17[0-8]|18[0-9]|19[0-35-9])\d{8}$/.test(val);
};
const isImgPath = (path) => {
  return /(https?:\/\/|data:image\/).*?\.(png|jpg|jpeg|gif|svg|webp|ico)/gi.test(path);
};
const isIOS = () => {
  return isClient && window?.navigator?.userAgent && (/iP(?:ad|hone|od)/.test(window.navigator.userAgent) || window?.navigator?.maxTouchPoints > 2 && /iPad|Macintosh/.test(window?.navigator.userAgent));
};
const isEmpty = (val, checkFull = true) => {
  if (isNumber(val) && isNaN(val)) return true;
  if (val === "" || val === null || val === void 0) return true;
  if (!checkFull) return false;
  if (isArray(val) && val.length === 0) return true;
  if (isObject(val) && Object.keys(val).length === 0) return true;
  return false;
};
const isFocusable = (element) => {
  if (element.tabIndex > 0 || element.tabIndex === 0 && element.getAttribute("tabIndex") !== null) {
    return true;
  }
  if (element.tabIndex < 0 || element.hasAttribute("disabled") || element.getAttribute("aria-disabled") === "true") {
    return false;
  }
  switch (element.nodeName) {
    case "A": {
      return !!element.href && element.rel !== "ignore";
    }
    case "INPUT": {
      return !(element.type === "hidden" || element.type === "file");
    }
    case "BUTTON":
    case "SELECT":
    case "TEXTAREA": {
      return true;
    }
    default: {
      return false;
    }
  }
};

exports.inBrowser = inBrowser;
exports.is = is;
exports.isArray = isArray;
exports.isAsyncFunction = isAsyncFunction;
exports.isBoolean = isBoolean;
exports.isClient = isClient;
exports.isDate = isDate;
exports.isDef = isDef;
exports.isElement = isElement;
exports.isEmpty = isEmpty;
exports.isExternal = isExternal;
exports.isFocusable = isFocusable;
exports.isFunction = isFunction;
exports.isIOS = isIOS;
exports.isImageDom = isImageDom;
exports.isImgPath = isImgPath;
exports.isNull = isNull;
exports.isNullAndUnDef = isNullAndUnDef;
exports.isNullOrUnDef = isNullOrUnDef;
exports.isNumber = isNumber;
exports.isObject = isObject;
exports.isPhone = isPhone;
exports.isPromise = isPromise;
exports.isServer = isServer;
exports.isString = isString;
exports.isStringNumber = isStringNumber;
exports.isType = isType;
exports.isUnDef = isUnDef;
exports.isValidURL = isValidURL;
