// vitepress-theme-teek/lib/components/theme/ThemeEnhance/src/themeEnhance.js
'use strict';

var LayoutMode = /* @__PURE__ */ ((LayoutMode2) => {
  LayoutMode2["FullWidth"] = "fullWidth";
  LayoutMode2["SidebarWidthAdjustableOnly"] = "sidebarWidthAdjustableOnly";
  LayoutMode2["BothWidthAdjustable"] = "bothWidthAdjustable";
  LayoutMode2["Original"] = "original";
  return LayoutMode2;
})(LayoutMode || {});
var SpotlightStyle = /* @__PURE__ */ ((SpotlightStyle2) => {
  SpotlightStyle2["Under"] = "under";
  SpotlightStyle2["Aside"] = "aside";
  return SpotlightStyle2;
})(SpotlightStyle || {});
var ThemeColor = /* @__PURE__ */ ((ThemeColor2) => {
  ThemeColor2["vpDefault"] = "vp-default";
  ThemeColor2["vpGreen"] = "vp-green";
  ThemeColor2["vpYellow"] = "vp-yellow";
  ThemeColor2["vpRed"] = "vp-red";
  ThemeColor2["epBlue"] = "ep-blue";
  ThemeColor2["epGreen"] = "ep-green";
  ThemeColor2["epYellow"] = "ep-yellow";
  ThemeColor2["epRed"] = "ep-red";
  return ThemeColor2;
})(ThemeColor || {});
const themeColorList = [
  "vp-default" /* vpDefault */,
  "vp-green" /* vpGreen */,
  "vp-yellow" /* vpYellow */,
  "vp-red" /* vpRed */,
  "ep-blue" /* epBlue */,
  "ep-green" /* epGreen */,
  "ep-yellow" /* epYellow */,
  "ep-red" /* epRed */
];
const mobileMaxWidthMedia = "(max-width: 768px)";
const activateMaxWidthSlideMedia = "(min-width: 1440px)";
const touchMedia = "(pointer: coarse)";
const layoutModeAttribute = "layout-mode";
const themeColorAttribute = "theme-color";

exports.LayoutMode = LayoutMode;
exports.SpotlightStyle = SpotlightStyle;
exports.ThemeColor = ThemeColor;
exports.activateMaxWidthSlideMedia = activateMaxWidthSlideMedia;
exports.layoutModeAttribute = layoutModeAttribute;
exports.mobileMaxWidthMedia = mobileMaxWidthMedia;
exports.themeColorAttribute = themeColorAttribute;
exports.themeColorList = themeColorList;
exports.touchMedia = touchMedia;
