// vitepress-plugin-permalink/dist/index.cjs
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const node_fs = require('node:fs');
const node_path = require('node:path');
const matter = require('gray-matter');
const node_url = require('node:url');
const vite = require('vite');
const picocolors = require('picocolors');

var _documentCurrentScript = typeof document !== 'undefined' ? document.currentScript : null;
function _interopDefaultCompat (e) { return e && typeof e === 'object' && 'default' in e ? e.default : e; }

const matter__default = /*#__PURE__*/_interopDefaultCompat(matter);
const picocolors__default = /*#__PURE__*/_interopDefaultCompat(picocolors);

const DEFAULT_IGNORE_DIR = ["node_modules", "dist", ".vitepress", "public"];
const permalinks = {};
const createPermalinks = (option = {}, cleanUrls = false) => {
  const { path, ignoreList = [] } = option;
  if (!path) return {};
  const dirPaths = readDirPaths(path, ignoreList);
  scannerMdFile(path, option, "", cleanUrls, true);
  dirPaths.forEach((dirPath) => scannerMdFile(dirPath, option, node_path.basename(dirPath), cleanUrls));
  return permalinks;
};
const readDirPaths = (sourceDir, ignoreList = []) => {
  const dirPaths = [];
  const ignoreListAll = [...DEFAULT_IGNORE_DIR, ...ignoreList];
  const secondDirNames = node_fs.readdirSync(sourceDir);
  secondDirNames.forEach((secondDirName) => {
    const secondDirPath = node_path.resolve(sourceDir, secondDirName);
    if (!isSome(ignoreListAll, secondDirName) && node_fs.statSync(secondDirPath).isDirectory()) {
      dirPaths.push(secondDirPath);
    }
  });
  return dirPaths;
};
const scannerMdFile = (root, option, prefix = "", cleanUrls = false, onlyScannerRootMd = false) => {
  const { ignoreList = [] } = option;
  const ignoreListAll = [...DEFAULT_IGNORE_DIR, ...ignoreList];
  const secondDirOrFilenames = node_fs.readdirSync(root);
  secondDirOrFilenames.forEach((dirOrFilename) => {
    if (isSome(ignoreListAll, dirOrFilename)) return;
    const filePath = node_path.resolve(root, dirOrFilename);
    if (!onlyScannerRootMd && node_fs.statSync(filePath).isDirectory()) {
      scannerMdFile(filePath, option, `${prefix}/${dirOrFilename}`, cleanUrls);
    } else {
      if (!isMdFile(dirOrFilename)) return;
      const content = node_fs.readFileSync(filePath, "utf-8");
      const { data: { permalink = "" } = {} } = matter__default(content, {});
      if (permalink) {
        const filename = node_path.basename(dirOrFilename, node_path.extname(dirOrFilename));
        const finalPermalink = standardLink(permalink);
        permalinks[`${prefix ? `${prefix}/` : ""}${filename}`] = cleanUrls ? finalPermalink : `${finalPermalink}.html`;
      }
    }
  });
};
const isMdFile = (filePath) => {
  return filePath.includes("md") || filePath.includes("MD");
};
const isSome = (arr, name) => {
  return arr.some((item) => item === name || item instanceof RegExp && item.test(name));
};
const standardLink = (permalink = "") => {
  let finalPermalink = permalink;
  if (!finalPermalink.startsWith("/")) finalPermalink = "/" + finalPermalink;
  if (finalPermalink.endsWith("/")) finalPermalink = finalPermalink.replace(/\/$/, "");
  return finalPermalink;
};

const version = "1.1.2";

const logger = vite.createLogger("info", {
  prefix: `[vitepress-plugin-permalink v${version}]`
});
const info = (message, level = "green", option = { timestamp: true }) => {
  logger.info(picocolors__default[level](message), option);
};
const warn = (message, level = "yellow", option = { timestamp: true }) => {
  logger.warn(picocolors__default[level](message), option);
};
const warnOnce = (message, level = "yellow", option = { timestamp: true }) => {
  logger.info(picocolors__default[level](message), option);
};
const error = (message, level = "red", option = { timestamp: true }) => {
  logger.error(picocolors__default[level](message), option);
};
const logger$1 = {
  info,
  warn,
  warnOnce,
  error
};

function VitePluginVitePressPermalink(option = {}) {
  return [
    VitePluginVitePressAutoPermalink(option.permalinkOption),
    VitePluginVitePressUsePermalink(option.notFoundOption)
  ];
}
function VitePluginVitePressAutoPermalink(option = {}) {
  let isExecute = false;
  let vitepressConfig = {};
  return {
    name: "vite-plugin-vitepress-auto-permalink",
    config(config) {
      if (isExecute) return;
      isExecute = true;
      const {
        site: { themeConfig, cleanUrls, locales },
        srcDir,
        rewrites
      } = config.vitepress;
      const baseDir = option.path ? node_path.join(srcDir, option.path) : srcDir;
      const permalinks = createPermalinks({ ...option, path: baseDir }, cleanUrls);
      const pathToPermalink = {};
      const permalinkToPath = {};
      const localesKeys = Object.keys(locales || {});
      for (const [key, value] of Object.entries(permalinks)) {
        const rewriteFilePath = rewrites.map[`${key}.md`]?.replace(/\.md/, "") || key;
        const newValue = getLocalePermalink(localesKeys, key, value);
        if (permalinkToPath[newValue]) {
          logger$1.warn(`\u6C38\u4E45\u94FE\u63A5 '${newValue}' \u5DF2\u5B58\u5728\uFF0C\u5176\u5BF9\u5E94\u7684 '${permalinkToPath[newValue]}' \u5C06\u4F1A\u88AB '${key}' \u8986\u76D6`);
        }
        pathToPermalink[rewriteFilePath] = newValue;
        permalinkToPath[newValue] = rewriteFilePath;
      }
      themeConfig.permalinks = { map: pathToPermalink, inv: permalinkToPath };
      logger$1.info("Injected Permalinks Data Successfully. \u6CE8\u5165\u6C38\u4E45\u94FE\u63A5\u6570\u636E\u6210\u529F!");
      vitepressConfig = config.vitepress;
      if (!localesKeys.length) {
        return setActiveMatchWhenUsePermalink({
          nav: themeConfig.nav,
          permalinkToPath,
          rewrites,
          cleanUrls
        });
      }
      localesKeys.forEach((localeKey) => {
        setActiveMatchWhenUsePermalink({
          nav: locales[localeKey].themeConfig?.nav,
          permalinkToPath,
          rewrites,
          cleanUrls
        });
      });
    },
    // 仅限 dev 环境生效
    configureServer(server) {
      const {
        site: {
          base,
          themeConfig: { permalinks }
        },
        rewrites
      } = vitepressConfig;
      server.middlewares.use((req, _res, next) => {
        if (req.url && req.url.includes(".md")) {
          const reqUrl = decodeURI(req.url).replace(/[?#].*$/, "").replace(/\.md$/, "").slice(base.length);
          const path = "/" + reqUrl.replace(/^\//, "");
          const filePath = permalinks.inv[path] || permalinks.inv[`${path}.html`];
          const realFilePath = rewrites.inv[`${filePath}.md`]?.replace(/\.md/, "") || filePath;
          if (realFilePath) req.url = req.url.replace(encodeURI(reqUrl), encodeURI(realFilePath));
        }
        next();
      });
    }
  };
}
const getLocalePermalink = (localesKeys = [], path = "", permalink = "") => {
  const localesKey = localesKeys.filter((key) => key !== "root").find((key) => path.startsWith(key));
  if (localesKey) return `/${localesKey}${permalink.startsWith("/") ? permalink : `/${permalink}`}`;
  return permalink;
};
const setActiveMatchWhenUsePermalink = (option, parentNav) => {
  const { nav = [], permalinkToPath, rewrites = {}, cleanUrls = false } = option;
  if (!nav.length) return;
  nav.forEach((item) => {
    if (item.link === "/" || item.activeMatch) return;
    const link = standardLink(item.link);
    const path = permalinkToPath[cleanUrls ? link : `${link.replace(/\.html/, "")}.html`];
    if (path) {
      const finalPath = rewrites.map[`${path}.md`]?.replace(/\.md/, "") || path;
      item.activeMatch = finalPath;
      if (parentNav) parentNav.activeMatch = finalPath.slice(0, finalPath.lastIndexOf("/"));
    }
    if (item.items?.length) {
      setActiveMatchWhenUsePermalink({ nav: item.items, permalinkToPath, rewrites, cleanUrls }, item);
    }
  });
};
const isESM = () => {
  return typeof __filename === "undefined" || typeof __dirname === "undefined";
};
const getDirname = () => {
  return isESM() ? node_path.dirname(node_url.fileURLToPath((typeof document === 'undefined' ? require('u' + 'rl').pathToFileURL(__filename).href : (_documentCurrentScript && _documentCurrentScript.tagName.toUpperCase() === 'SCRIPT' && _documentCurrentScript.src || new URL('index.cjs', document.baseURI).href)))) : __dirname;
};
function VitePluginVitePressUsePermalink(option = {}) {
  const usePermalinkFile = `usePermalink`;
  const aliasUsePermalinkFile = `${getDirname()}/${usePermalinkFile}`;
  const NotFoundDelayComponentName = "NotFoundDelay";
  const NotFoundDelayComponentFile = `${NotFoundDelayComponentName}.vue`;
  const aliasNotFoundDelayComponentFile = `${getDirname()}/components/${NotFoundDelayComponentFile}`;
  const virtualModuleId = "virtual:not-found-option";
  const resolvedVirtualModuleId = `\0${virtualModuleId}`;
  return {
    name: "vite-plugin-vitepress-use-permalink",
    config() {
      return {
        resolve: {
          alias: {
            [`./${usePermalinkFile}`]: aliasUsePermalinkFile,
            [`./${NotFoundDelayComponentFile}`]: aliasNotFoundDelayComponentFile
          }
        }
      };
    },
    resolveId(id) {
      if (id === virtualModuleId) return resolvedVirtualModuleId;
    },
    load(id) {
      if (id === resolvedVirtualModuleId) return `export default ${JSON.stringify(option)}`;
      if (id.endsWith("vitepress/dist/client/theme-default/Layout.vue")) {
        const code = node_fs.readFileSync(id, "utf-8");
        const slotName = "not-found";
        const notFoundSlotPosition = `<slot name="${slotName}" />`;
        const setupPosition = '<script setup lang="ts">';
        return code.replace(
          notFoundSlotPosition,
          `<${NotFoundDelayComponentName}><template #${slotName}>${notFoundSlotPosition}</template></${NotFoundDelayComponentName}>`
        ).replace(
          setupPosition,
          `${setupPosition}
import ${usePermalinkFile} from './${usePermalinkFile}'
import ${NotFoundDelayComponentName} from './${NotFoundDelayComponentFile}'`
        ).replace("<\/script>", `${usePermalinkFile}().startWatch() <\/script>`);
      }
    }
  };
}

exports.VitePluginVitePressAutoPermalink = VitePluginVitePressAutoPermalink;
exports.VitePluginVitePressUsePermalink = VitePluginVitePressUsePermalink;
exports.default = VitePluginVitePressPermalink;
