// vitepress-theme-teek/lib/components/theme/HomeMyCard/src/index.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var vitepress = require('vitepress');
var useLocale = require('../../../../composables/useLocale.js');
var useNamespace = require('../../../../composables/useNamespace.js');
require('../../../../composables/useZIndex.js');
var index = require('../../ConfigProvider/index.js');
var index_vue_vue_type_script_setup_true_name_PageCard_lang = require('../../../common/PageCard/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_Avatar_lang = require('../../../common/Avatar/src/index.vue2.js');
var index_vue_vue_type_script_setup_true_name_Icon_lang = require('../../../common/Icon/src/index.vue2.js');
require('@iconify/vue');

const _hoisted_1 = ["aria-label"];
const _hoisted_2 = ["href", "title", "aria-label"];
const _hoisted_3 = ["aria-label"];
const _hoisted_4 = { class: "name" };
const _hoisted_5 = { class: "slogan" };
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "HomeMyCard" },
  __name: "index",
  setup(__props) {
    const ns = useNamespace.useNamespace("my");
    const { t } = useLocale.useLocale();
    const { getTeekConfigRef } = index.useTeekConfig();
    const blogger = getTeekConfigRef("blogger", { shape: "square", circleBgMask: true });
    const social = getTeekConfigRef("social", []);
    const shape = Vue.computed(() => blogger.value.shape.replace(/-.*$/, ""));
    const isCircleBgImg = Vue.computed(() => shape.value === "circle" && !!blogger.value.circleBgImg);
    const avatarBgStyle = Vue.computed(() => ({ backgroundImage: `url(${vitepress.withBase(blogger.value.circleBgImg)})` }));
    const myCardColorStyle = Vue.computed(() => ({ color: blogger.value.color }));
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createElementBlock(
        Vue.Fragment,
        null,
        [
          Vue.renderSlot(_ctx.$slots, "teek-home-card-my-before"),
          Vue.renderSlot(_ctx.$slots, "teek-home-card-my", {}, () => [
            Vue.unref(blogger).name ? (Vue.openBlock(), Vue.createBlock(Vue.unref(index_vue_vue_type_script_setup_true_name_PageCard_lang.default), {
              key: 0,
              class: Vue.normalizeClass([Vue.unref(ns).b(), Vue.unref(ns).is("circle-bg", isCircleBgImg.value)]),
              style: Vue.normalizeStyle(myCardColorStyle.value),
              "aria-label": Vue.unref(t)("tk.myCard.label")
            }, {
              default: Vue.withCtx(() => [
                isCircleBgImg.value ? (Vue.openBlock(), Vue.createElementBlock(
                  "div",
                  {
                    key: 0,
                    class: Vue.normalizeClass([Vue.unref(ns).em("avatar__circle", "bg"), Vue.unref(ns).is("mask", Vue.unref(blogger).circleBgMask)]),
                    style: Vue.normalizeStyle(avatarBgStyle.value)
                  },
                  null,
                  6
                  /* CLASS, STYLE */
                )) : Vue.createCommentVNode("v-if", true),
                Vue.createElementVNode(
                  "div",
                  {
                    class: Vue.normalizeClass(`${Vue.unref(ns).e("avatar")} ${Vue.unref(blogger).shape} flx-center`)
                  },
                  [
                    Vue.unref(blogger).avatar ? (Vue.openBlock(), Vue.createBlock(Vue.unref(index_vue_vue_type_script_setup_true_name_Avatar_lang.default), {
                      key: 0,
                      src: Vue.unref(vitepress.withBase)(Vue.unref(blogger).avatar),
                      size: Vue.unref(blogger).shape === "square" ? "100%" : 100,
                      shape: shape.value,
                      "bg-color": "transparent",
                      alt: Vue.unref(t)("tk.myCard.avatarAlt"),
                      title: Vue.unref(t)("tk.myCard.avatarTitle"),
                      "aria-hidden": "true"
                    }, null, 8, ["src", "size", "shape", "alt", "title"])) : (Vue.openBlock(), Vue.createBlock(Vue.unref(index_vue_vue_type_script_setup_true_name_Avatar_lang.default), {
                      key: 1,
                      size: 100,
                      shape: shape.value,
                      text: Vue.unref(blogger).name,
                      "text-size": 50,
                      "bg-color": Vue.unref(ns).cssVar("theme-color"),
                      "aria-hidden": "true"
                    }, null, 8, ["shape", "text", "bg-color"]))
                  ],
                  2
                  /* CLASS */
                ),
                Vue.unref(social).length ? (Vue.openBlock(), Vue.createElementBlock("div", {
                  key: 1,
                  class: Vue.normalizeClass(`${Vue.unref(ns).e("icons")} flx-justify-around`),
                  "aria-label": Vue.unref(t)("tk.myCard.socialLabel")
                }, [
                  (Vue.openBlock(true), Vue.createElementBlock(
                    Vue.Fragment,
                    null,
                    Vue.renderList(Vue.unref(social), (item, index) => {
                      return Vue.openBlock(), Vue.createElementBlock("a", {
                        key: index,
                        href: item.link && Vue.unref(vitepress.withBase)(item.link),
                        title: item.name,
                        target: "_blank",
                        "aria-label": item.name
                      }, [
                        item.icon ? (Vue.openBlock(), Vue.createBlock(Vue.unref(index_vue_vue_type_script_setup_true_name_Icon_lang.default), {
                          key: 0,
                          iconType: item.iconType,
                          icon: item.icon,
                          size: "20px",
                          hover: "",
                          imgAlt: item.imgAlt,
                          "aria-hidden": "true"
                        }, null, 8, ["iconType", "icon", "imgAlt"])) : Vue.createCommentVNode("v-if", true)
                      ], 8, _hoisted_2);
                    }),
                    128
                    /* KEYED_FRAGMENT */
                  ))
                ], 10, _hoisted_1)) : Vue.createCommentVNode("v-if", true),
                Vue.createElementVNode("div", {
                  class: Vue.normalizeClass(Vue.unref(ns).e("blogger")),
                  "aria-label": Vue.unref(t)("tk.myCard.bloggerLabel")
                }, [
                  Vue.createElementVNode(
                    "h3",
                    _hoisted_4,
                    Vue.toDisplayString(Vue.unref(blogger).name),
                    1
                    /* TEXT */
                  ),
                  Vue.createElementVNode(
                    "span",
                    _hoisted_5,
                    Vue.toDisplayString(Vue.unref(blogger).slogan),
                    1
                    /* TEXT */
                  )
                ], 10, _hoisted_3)
              ]),
              _: 1
              /* STABLE */
            }, 8, ["class", "style", "aria-label"])) : Vue.createCommentVNode("v-if", true)
          ]),
          Vue.renderSlot(_ctx.$slots, "teek-home-card-my-after")
        ],
        64
        /* STABLE_FRAGMENT */
      );
    };
  }
});

exports.default = _sfc_main;
