// vitepress-theme-teek/lib/components/theme/HomeBanner/src/HomeBannerContent.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var vitepress = require('vitepress');
var useLocale = require('../../../../composables/useLocale.js');
var useNamespace = require('../../../../composables/useNamespace.js');
var useSwitchData = require('../../../../composables/useSwitchData.js');
var useTextTypes = require('../../../../composables/useTextTypes.js');
require('../../../../composables/useZIndex.js');
var index = require('../../ConfigProvider/index.js');

const _hoisted_1 = ["aria-label"];
const _hoisted_2 = ["aria-label"];
const _hoisted_3 = ["aria-label"];
const _hoisted_4 = { key: 0 };
const _hoisted_5 = ["aria-label"];
const _hoisted_6 = ["aria-label"];
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "HomeBannerContent" },
  __name: "HomeBannerContent",
  setup(__props) {
    const ns = useNamespace.useNamespace("banner-content");
    const { t } = useLocale.useLocale();
    const { getTeekConfigRef } = index.useTeekConfig();
    const { site, frontmatter } = vitepress.useData();
    const bannerConfig = getTeekConfigRef("banner", {
      name: frontmatter.value.tk?.name || site.value.title || "",
      descStyle: "default",
      description: [],
      switchTime: 4e3,
      switchShuffle: false,
      typesInTime: 200,
      typesOutTime: 100,
      typesNextTime: 800,
      typesShuffle: false
    });
    const descArray = Vue.computed(() => [
      ...new Set(
        [frontmatter.value.tk?.description || bannerConfig.value.description || []].flat()?.filter((v) => !!v)
      )
    ]);
    const isDefaultDescStyle = Vue.computed(() => bannerConfig.value.descStyle === "default");
    const isTypesDescStyle = Vue.computed(() => bannerConfig.value.descStyle === "types");
    const isSwitchDescStyle = Vue.computed(() => bannerConfig.value.descStyle === "switch");
    const {
      text: typesText,
      isFinished,
      start: startTypes
    } = useTextTypes.useTextTypes(descArray, {
      inputTime: bannerConfig.value.typesInTime,
      outputTime: bannerConfig.value.typesOutTime,
      nextTime: bannerConfig.value.typesNextTime,
      shuffle: bannerConfig.value.typesShuffle
    });
    const { data: text, start: startAutoSwitch } = useSwitchData.useSwitchData(descArray, {
      timeout: bannerConfig.value.switchTime,
      shuffle: bannerConfig.value.switchShuffle,
      onUpdate: (data, newValue) => {
        data.value = "";
        setTimeout(() => {
          data.value = newValue;
        }, 100);
      }
    });
    Vue.onMounted(() => {
      if (isTypesDescStyle.value) startTypes();
      if (isSwitchDescStyle.value) startAutoSwitch();
    });
    return (_ctx, _cache) => {
      return Vue.openBlock(), Vue.createElementBlock("div", {
        class: Vue.normalizeClass(Vue.unref(ns).b()),
        "aria-label": Vue.unref(t)("tk.homeBanner.contentLabel")
      }, [
        Vue.createElementVNode("h1", {
          class: Vue.normalizeClass(Vue.unref(ns).e("content__title")),
          "aria-label": Vue.unref(t)("tk.homeBanner.titleLabel")
        }, Vue.toDisplayString(Vue.unref(bannerConfig).name), 11, _hoisted_2),
        Vue.createElementVNode("p", {
          class: Vue.normalizeClass(Vue.unref(ns).e("content__desc")),
          "aria-label": Vue.unref(t)("tk.homeBanner.descLabel")
        }, [
          isDefaultDescStyle.value ? (Vue.openBlock(), Vue.createElementBlock(
            "span",
            _hoisted_4,
            Vue.toDisplayString(descArray.value[0]),
            1
            /* TEXT */
          )) : isSwitchDescStyle.value ? Vue.withDirectives((Vue.openBlock(), Vue.createElementBlock("span", {
            key: 1,
            onClick: _cache[0] || (_cache[0] = //@ts-ignore
            (...args) => Vue.unref(startAutoSwitch) && Vue.unref(startAutoSwitch)(...args)),
            class: "switch",
            "aria-label": Vue.unref(t)("tk.homeBanner.descSwitchLabel")
          }, Vue.toDisplayString(Vue.unref(text)), 9, _hoisted_5)), [
            [Vue.vShow, !!Vue.unref(text)]
          ]) : isTypesDescStyle.value && descArray.value.length ? (Vue.openBlock(), Vue.createElementBlock(
            Vue.Fragment,
            { key: 2 },
            [
              Vue.createElementVNode("span", {
                "aria-label": Vue.unref(t)("tk.homeBanner.descTypedLabel")
              }, Vue.toDisplayString(Vue.unref(typesText)), 9, _hoisted_6),
              Vue.createElementVNode(
                "span",
                {
                  class: Vue.normalizeClass(["typed", { "is-animation": Vue.unref(isFinished) }])
                },
                "|",
                2
                /* CLASS */
              )
            ],
            64
            /* STABLE_FRAGMENT */
          )) : Vue.createCommentVNode("v-if", true)
        ], 10, _hoisted_3)
      ], 10, _hoisted_1);
    };
  }
});

exports.default = _sfc_main;
