// vitepress-theme-teek/lib/components/theme/HomeBanner/src/HomeBannerFeature.vue2.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var Vue = require('vue');
var vitepress = require('vitepress');
var useLocale = require('../../../../composables/useLocale.js');
var useNamespace = require('../../../../composables/useNamespace.js');
var useWindowSize = require('../../../../composables/useWindowSize.js');
require('../../../../composables/useZIndex.js');
var index = require('../../ConfigProvider/index.js');

const _hoisted_1 = ["aria-labelledby"];
const _hoisted_2 = ["href", "aria-label"];
const _hoisted_3 = ["src", "alt", "aria-label"];
const _hoisted_4 = ["id"];
const _hoisted_5 = { class: "feature-description" };
var _sfc_main = /* @__PURE__ */ Vue.defineComponent({
  ...{ name: "HomeBannerFeature" },
  __name: "HomeBannerFeature",
  setup(__props) {
    const ns = useNamespace.useNamespace("banner-feature");
    const { t } = useLocale.useLocale();
    const { getTeekConfigRef } = index.useTeekConfig();
    const { frontmatter } = vitepress.useData();
    const bannerConfig = getTeekConfigRef("banner", {
      features: frontmatter.value.tk?.features || [],
      featureCarousel: 4e3
    });
    const active = Vue.ref(0);
    const isMobile = Vue.ref(false);
    let timer;
    const clearTimer = () => {
      if (timer) {
        clearTimeout(timer);
        timer = null;
      }
    };
    useWindowSize.useWindowSize((width) => {
      const { features, featureCarousel } = bannerConfig.value;
      if (width <= 719) {
        isMobile.value = true;
        clearTimer();
        timer = setInterval(() => {
          active.value = (active.value + 1) % features.length;
        }, featureCarousel);
      } else {
        isMobile.value = false;
        clearTimer();
      }
    });
    Vue.onBeforeUnmount(() => {
      clearTimer();
    });
    return (_ctx, _cache) => {
      return Vue.unref(bannerConfig).features.length ? (Vue.openBlock(), Vue.createBlock(Vue.TransitionGroup, {
        key: 0,
        name: Vue.unref(ns).joinNamespace("slide-next"),
        tag: "div",
        class: Vue.normalizeClass([Vue.unref(ns).b(), Vue.unref(ns).joinNamespace("wallpaper-outside"), "flx-wrap-between"]),
        "aria-label": Vue.unref(t)("tk.homeBanner.featureLabel")
      }, {
        default: Vue.withCtx(() => [
          (Vue.openBlock(true), Vue.createElementBlock(
            Vue.Fragment,
            null,
            Vue.renderList(Vue.unref(bannerConfig).features, (feature, index) => {
              return Vue.withDirectives((Vue.openBlock(), Vue.createElementBlock("div", {
                class: Vue.normalizeClass(Vue.unref(ns).e("feature__item")),
                key: index,
                role: "group",
                "aria-labelledby": `feature-title-${index}`
              }, [
                feature.link ? (Vue.openBlock(), Vue.createElementBlock("a", {
                  key: 0,
                  href: Vue.unref(vitepress.withBase)(feature.link),
                  class: "flx-column-center hover-color",
                  "aria-label": feature.title
                }, [
                  feature.imgUrl ? (Vue.openBlock(), Vue.createElementBlock("img", {
                    key: 0,
                    class: "feature-img",
                    src: Vue.unref(vitepress.withBase)(feature.imgUrl),
                    alt: feature.title,
                    "aria-label": feature.title
                  }, null, 8, _hoisted_3)) : Vue.createCommentVNode("v-if", true),
                  Vue.createElementVNode("p", {
                    id: `feature-title-${index}`,
                    class: "feature-title"
                  }, Vue.toDisplayString(feature.title), 9, _hoisted_4),
                  Vue.createElementVNode(
                    "p",
                    _hoisted_5,
                    Vue.toDisplayString(feature.description),
                    1
                    /* TEXT */
                  )
                ], 8, _hoisted_2)) : Vue.createCommentVNode("v-if", true)
              ], 10, _hoisted_1)), [
                [Vue.vShow, !isMobile.value || active.value === index]
              ]);
            }),
            128
            /* KEYED_FRAGMENT */
          ))
        ]),
        _: 1
        /* STABLE */
      }, 8, ["name", "class", "aria-label"])) : Vue.createCommentVNode("v-if", true);
    };
  }
});

exports.default = _sfc_main;
